authzed/api/v1/permission_service.proto

```diff
@@ -172,16 +172,36 @@ message SubjectFilter {
 // relationships within the system.
 message ReadRelationshipsRequest {
   Consistency consistency = 1;
+
+  // relationship_filter defines the filter to be applied to the relationships
+  // to be returned.
   RelationshipFilter relationship_filter = 2
       [ (validate.rules).message.required = true ];
+
+  // optional_limit, if non-zero, specifies the limit on the number of relationships to return
+  // before the stream is closed on the server side. By default, the stream will continue
+  // resolving relationships until exhausted or the stream is closed due to the client or a
+  // network issue.
+  uint32 optional_limit = 3 [(validate.rules).uint32 = {gte:0, lte: 1000}];
```

Comment:
We could allow any positive value for a limit, but not guarantee that it's used (e.g. user sets 10k, server returns 1k).